packages/legend-query/src/stores/postFilterOperators/QueryBuilderPostFilterOperator_Equal.ts

```diff
@@ -50,7 +53,37 @@ export class QueryBuilderPostFilterOperator_Equal extends QueryBuilderPostFilter
     return 'is';
   }
 
-  getPureFunction(): SUPPORTED_FUNCTIONS {
+  getPureFunction(
+    filterConditionState: PostFilterConditionState,
+  ): SUPPORTED_FUNCTIONS {
+    if (
+      filterConditionState.columnState.getReturnType()?.path ===
+        PRIMITIVE_TYPE.DATETIME &&
+      filterConditionState.value?.genericType?.value.rawType.path !==
+        PRIMITIVE_TYPE.DATETIME
+    ) {
+      return SUPPORTED_FUNCTIONS.IS_ON_DAY;
+    }
+    return SUPPORTED_FUNCTIONS.EQUAL;
+  }
+
+  getPureFunctionFromExpression(
+    expression: FunctionExpression,
+  ): SUPPORTED_FUNCTIONS | undefined {
+    let innerExpression = expression;
+    if (matchFunctionName(expression.functionName, SUPPORTED_FUNCTIONS.NOT)) {
+      innerExpression = guaranteeNonNullable(unwrapNotExpression(expression));
+    }
+    if (
+      innerExpression.parametersValues[0] instanceof
+        AbstractPropertyExpression &&
+      innerExpression.parametersValues[0].func.genericType.value.rawType
+        .path === PRIMITIVE_TYPE.DATETIME &&
+      innerExpression.parametersValues[1]?.genericType?.value.rawType.path !==
```

Comment:
The high-level idea here is the same as what we do to the filter.
Since Date is the supertype, RHS could be date/datetime/strictdate, etc.
For DateTime, if RHS is date/stricdate, isOnDay() needs to be called.